File: test/extensions/language/default_values/cpp/OptionalDefaultValuesTest.cpp
#include "default_values/optional_default_values/OptionalDefaultValues.h"
#include "gtest/gtest.h"
#include "test_utils/TestUtility.h"

namespace default_values
{
namespace optional_default_values
{

TEST(OptionalDefaultValuesTest, checkOptionalNoDefaultBoolField)
{
    OptionalDefaultValues data;
    ASSERT_FALSE(data.isOptionalNoDefaultBoolFieldSet());
}

TEST(OptionalDefaultValuesTest, checkOptionalNoDefaultStringField)
{
    OptionalDefaultValues data;
    ASSERT_FALSE(data.isOptionalNoDefaultStringFieldSet());
}

TEST(OptionalDefaultValuesTest, checkOptionalDefaultU32Field)
{
    OptionalDefaultValues data;
    ASSERT_TRUE(data.isOptionalDefaultU32FieldSet());
    ASSERT_EQ(13, data.getOptionalDefaultU32Field());
}

TEST(OptionalDefaultValuesTest, checkOptionalDefaultF64Field)
{
    OptionalDefaultValues data;
    ASSERT_TRUE(data.isOptionalDefaultF64FieldSet());
    ASSERT_EQ(1.234, data.getOptionalDefaultF64Field());
}

TEST(OptionalDefaultValuesTest, checkOptionalDefaultStringField)
{
    OptionalDefaultValues data;
    ASSERT_TRUE(data.isOptionalDefaultStringFieldSet());
    ASSERT_EQ("default", data.getOptionalDefaultStringField());
}

TEST(OptionalDefaultValuesTest, writeRead)
{
    OptionalDefaultValues data;
    test_utils::writeReadTest(data);
}

} // namespace optional_default_values
} // namespace default_values

File: test/extensions/language/default_values/python/OptionalDefaultValuesTest.py
import DefaultValues

from compoundutils import writeReadTest


class OptionalDefaultValuesTest(DefaultValues.TestCase):
    def testOptionalNoDefaultBoolField(self):
        data = self.api.OptionalDefaultValues()
        self.assertEqual(False, data.is_optional_no_default_bool_field_set())

    def testOptionalNoDefaultString(self):
        data = self.api.OptionalDefaultValues()
        self.assertEqual(False, data.is_optional_no_default_string_field_set())

    def testOptionalDefaultU32(self):
        data = self.api.OptionalDefaultValues()
        self.assertEqual(True, data.is_optional_default_u32_field_set())
        self.assertEqual(13, data.optional_default_u32_field)

    def testOptionalDefaultF64(self):
        data = self.api.OptionalDefaultValues()
        self.assertEqual(True, data.is_optional_default_f64_field_set())
        self.assertEqual(1.234, data.optional_default_f64_field)

    def testOptionalDefaultString(self):
        data = self.api.OptionalDefaultValues()
        self.assertEqual(True, data.is_optional_default_string_field_set())
        self.assertEqual("default", data.optional_default_string_field)

    def testwriteRead(self):
        data = self.api.OptionalDefaultValues()
        writeReadTest(self.api.OptionalDefaultValues, data)
